Neos.Utility.Schema/Classes/TYPO3/Flow/Utility/SchemaValidator.php

```diff
@@ -58,18 +68,42 @@ class SchemaValidator
      *
      * @param mixed $value value to validate
      * @param mixed $schema type as string, schema or array of schemas
+     * @param array $types the additional type schemas
      * @return \TYPO3\Flow\Error\Result
      */
-    public function validate($value, $schema)
+    public function validate($value, $schema, array $types = [])
     {
         $result = new \TYPO3\Flow\Error\Result();
+        if (is_array($schema)) {
+            foreach ($schema as $possibleTypeKey => $possibleTypeSchema) {
+                if (preg_match('/^@.*$/', $possibleTypeKey)) {
```

Comment:
Could be replaced with `$possibleTypeKey[0] === '@'` instead of the costly and harder to read preg_match